generators/generator-base-blueprint.js

```diff
@@ -84,51 +90,187 @@ module.exports = class extends BaseGenerator {
-                this.options.blueprint ||
-                this.configOptions.blueprint ||
-                this.config.get('blueprint')
-        );
-
-        if (blueprints && blueprints.length > 0) {
-            // Verify blueprints, should be executed only once
-            if (!this.configOptions.blueprintsVerified) {
-                // Run a lookup to find blueprints.
-                const packagePatterns = blueprints
-                    .filter(bp => !this.env.isPackageRegistered(jhipsterUtils.packageNameToNamespace(bp.name)))
-                    .map(bp => bp.name);
-                this.env.lookup({ filterPaths: true, packagePatterns });
-
-                if (!this.options.skipChecks) {
-                    const namespaces = blueprints.map(bp => jhipsterUtils.packageNameToNamespace(bp.name));
-                    // Verify if the blueprints has been registered.
-                    const missing = namespaces.filter(namespace => !this.env.isPackageRegistered(namespace));
-                    if (missing && missing.length > 0) {
-                        this.error(`Some blueprints were not found ${missing}, you should install them manually`);
-                    }
+        // Load and verify blueprints, should be executed only once
+        if (!this.configOptions.blueprintConfigured) {
+            this.configOptions.blueprintConfigured = true;
+            let argvBlueprints = this.options.blueprints || '';
+            // check for old single blueprint declaration
+            const blueprint = this.options.blueprint;
+            if (blueprint) {
+                this.warning('--blueprint option is deprecated. Please use --blueprints instead');
+                if (!argvBlueprints.split(',').includes(blueprint)) {
+                    argvBlueprints = `${blueprint},${argvBlueprints}`;
                 }
+            }
+            const blueprints = jhipsterUtils.mergeBlueprints(
+                jhipsterUtils.parseBluePrints(argvBlueprints),
+                jhipsterUtils.loadBlueprintsFromConfiguration(this.config)
+            );
+            // Run a lookup to find blueprints.
+            const packagePatterns = blueprints
+                .filter(bp => !this.env.isPackageRegistered(jhipsterUtils.packageNameToNamespace(bp.name)))
+                .map(bp => bp.name);
+            this.env.lookup({ filterPaths: true, packagePatterns });
 
-                this.configOptions.blueprintsVerified = true;
+            let otherModules = this.jhipsterConfig.otherModules || [];
+            if (blueprints && blueprints.length > 0) {
+                blueprints.forEach(blueprint => {
+                    blueprint.version = this.findBlueprintVersion(blueprint.name) || blueprint.version;
+                });
+
+                // Remove potential previous value to avoid duplicates
+                otherModules = otherModules.filter(module => this.blueprints.findIndex(bp => bp.name === module.name) === -1);
```

Comment:
if you need write comments, then it should be moved to another function with an explicit name :)

Reply:
I don't agree.
Too many methods makes less linear and more complicated.
The alternative is to let the developer guess or look at the api documentation.

Reply:
>  Too many methods makes less linear and more complicated.

Yup, and that's when one can see if the private code really belongs in its current place or if it's to be moved elsewhere.
My point is more about clean code than something else here. A function/method should be:
  - well named
  - short so as to understand what it does according to its name, around 8-10 lines
  - etc.

A comment inside this kind of block sometimes is an indication of something that can be improved. For instance, this kind of comments should be kept for really complex code (that almost never happens, unless you're working on a new spacecraft reactor or something that has a visible algorithmic complexity).
If you deem the comment code is clear enough, then the comment isn't necessary. Otherwise, moving it should be an option to consider.
A better explanation than mine: https://refactoring.guru/smells/comments.

Reply:
`otherModules` is used by templates, I can’t see a reason for extracting into a function.
We probably should start merging blueprints and modules at load instead of before save.
It duplicating and is implemented wrong, otherModules version will take precedence, this is wrong, I fix that later.

Your comments to extract to methods were:
- One field that should be improved. But this PR is big enough.
- A external api that could be improved.
- One deprecated option.